tests/tests_environments.yml

```diff
@@ -0,0 +1,86 @@
+# SPDX-License-Identifier: MIT
+---
+- name: Basic repository enablement/disablement test
+  hosts: all
+  gather_facts: true
+  become: true
+  tags:
+    - tests::slow
+  tasks:
+    - name: Setup Candlepin
+      import_tasks: tasks/setup_candlepin.yml
+
+    - name: Skip if no test environments are set
+      meta: end_play
+      when:
+        - >-
+          lsr_rhc_test_data.envs_register is none
+          or lsr_rhc_test_data.envs_register | d([]) | length == 0
+
+    - name: Try to register (wrong environment)
+      block:
+        - name: Register (wrong environment)
+          include_role:
+            name: linux-system-roles.rhc
+          vars:
+            rhc_auth:
+              login:
+                username: "{{ lsr_rhc_test_data.reg_username }}"
+                password: "{{ lsr_rhc_test_data.reg_password }}"
+            rhc_insights:
+              state: absent
+            rhc_organization: "{{ lsr_rhc_test_data.reg_organization }}"
+            rhc_server:
+              hostname: "{{ lsr_rhc_test_data.candlepin_host }}"
+              port: "{{ lsr_rhc_test_data.candlepin_port }}"
+              prefix: "{{ lsr_rhc_test_data.candlepin_prefix }}"
+              insecure: "{{ lsr_rhc_test_data.candlepin_insecure }}"
+            rhc_baseurl: "{{ lsr_rhc_test_data.baseurl | d(omit) }}"
+            rhc_environments:
+              - "{{ lsr_rhc_test_data.env_nonworking }}"
+
+        - name: Unreachable task
+          fail:
+            msg: The above task must fail
+      rescue:
+        - name: Assert registration failed
+          assert:
+            that: true
```

Comment:
```suggestion
            that: ansible_failed_result.msg != 'The above task must fail'
```

Reply:
Hm I did the same mistake in other tests too -- #50